lib/url.py

```diff
@@ -66,29 +66,28 @@ def archive_message_url(
 
 ## String cleaning functions
 
-# remove non-alnum ascii symbols from string
 def sanitize(s):
-    return (
-        "".join(
-            filter(
-                lambda x: x.isalnum or x == " ",
-                s.encode("ascii", "ignore").decode("utf-8"),
-            )
-        )
-        .replace(" ", "-")
-        .replace("?", "%3F")
-    )
-
+    """
+    Sanitize the string to a safe string that can be used in URLs
 
-# create a unique sanitized identifier for a topic
-def sanitize_topic(topic_name):
+    This is copied from Zulip's core code:
+    https://github.com/zulip/zulip/blob/de31114d700561f32139a63a0e5f33d5c30039b3/zerver/lib/url_encoding.py#L8
+    """
     return (
-        urllib.parse.quote(topic_name, safe="~()*!.'")
```

Comment:
The "~()*!.'" seems to be arbitrary. It's from this commit: https://github.com/zulip/zulip-archive/commit/f8bea8a9cb55ac318009f07888e6bd510024679e